src/Api/Controller/AbstractSerializeController.php

```diff
@@ -139,6 +146,47 @@ abstract protected function data(ServerRequestInterface $request, Document $docu
      */
     abstract protected function createElement($data, SerializerInterface $serializer);
 
+    /**
+     * Eager loads the required relationships.
+     *
+     * @param Collection $models
+     * @param array $relations
+     * @return void
+     */
+    protected function loadRelations(Collection $models, array $relations): void
```

Comment:
Could this replace the special case on line 129 of ListDiscussionsController?

Reply:
You are referring to this ? https://github.com/flarum/core/blob/2009f790d9112a3a8705e02569ff407c953cc7db/src/Api/Controller/ListDiscussionsController.php#L128-L136

I suppose we could, but then it would mean more queries to re-load the discussions and their required relations (tags for instance) that'd be at least 6+ queries, even though they've already been loaded. Unless you meant having the method handle that situation, but that also sounds like more complexity.

Reply:
Let's keep it as an edge case for now then :+1: